beamer/chain.py

```diff
@@ -48,26 +51,26 @@ def wrapper(*args, **kwargs):  # type: ignore
     return wrapper
 
 
-class ContractEventMonitor:
+class EventMonitor:
     def __init__(
         self,
-        name: str,
-        contract: web3.contract.Contract,
+        web3: Web3,
+        contracts: list[Contract],
```

Comment:
I'd prefer if we used `tuple[Contract]` here -- mutability is not really required and tuples would be more robust against accidental changes to the list outside of the event monitor.

Reply:
Ho9w do I tell python that a tuple of arbitrary length is fine?